### ISSUE TITLE
[🐛 BUG] Scope issue with lambda function

### ISSUE THREAD

**Issue by FlorianJacta** [Avaiga/taipy#2242]
### What went wrong? 🤔

There are issues using the lambda functions with multiple pages.

```
C:\Users\jacta\.conda\envs\test\Lib\site-packages\taipy\gui\utils\_evaluator.py:369: TaipyGuiWarning: Exception raised evaluating __lambda_2466438897376_1_TPMDL_0(x):

name 'create_chart' is not defined
  _warn(f"Exception raised evaluating {expr_string}", e)

```

(Note that this issue was first seen when trying to have multiple states for an application; this might not be the same issue)

### Expected Behavior

This should work fine like the comments in the code example.

### Steps to Reproduce Issue

*app.py*

```python
from taipy import Gui
from page_py import page # if you add 'create_chart' in the import, then this works!

pages = {"/": "App", "page": page}

Gui(pages=pages).run()
```

*page_py.py*
```python
import taipy.gui.builder as tgb
import pandas as pd
import plotly.express as px

x = 0


def create_chart(x):
    print("Hello")
    df = pd.DataFrame(
        {
            "Fruit": ["Apples", "Oranges", "Bananas", "Apples", "Oranges", "Bananas"],
            "Amount": [4, 1, 2, 2, 4, 5],
            "City": ["SF", "SF", "SF", "Montreal", "Montreal", "Montreal"],
        }
    )
    fig = px.bar(df, x="Fruit", y="Amount", color="City", barmode="group")
    fig.show()
    return fig


with tgb.Page() as page:
    tgb.text("Hello World")
    tgb.slider("{x}")
    tgb.chart(figure=lambda x: create_chart(x))

    # If you use the Markdown syntax, there are no issues. This works
    # tgb.chart(figure="{create_chart(x)}")
```` 

### Browsers

Chrome

### OS

Windows

### Version of Taipy

4.0.1

### Acceptance Criteria

- [ ] A unit test reproducing the bug is added.
- [ ] Any new code is covered by a unit tested.
- [ ] Check code coverage is at least 90%.
- [ ] The bug reporter validated the fix.
- [ ] Related issue(s) in taipy-doc are created for documentation and Release Notes are updated.

### Code of Conduct

- [X] I have checked the [existing issues](https://github.com/Avaiga/taipy/issues?q=is%3Aissue+).
- [ ] I am willing to work on this issue (optional)

**Comment by FlorianJacta** [comment]
Should this be backported?

**Comment by FredLL-Avaiga** [comment]
> Should this be backported?

Was it supported and documented in 3.1?
Is it a regression?

**Comment by FlorianJacta** [comment]
Is that the criteria to be backported?

It was working in 4.0.dev1 for example; I don't know if this is called a regression. Zacch had the same issue when writing its application

**Comment by FlorianJacta** [comment]
Fixed in 4.0.2.dev1. I will test if this fixes my original issue
